auto_tune_vllm/benchmarks/providers.py

```diff
@@ -35,27 +40,126 @@ def set_trial_context(self, study_name: str, trial_id: str):
         }
     
     def terminate_benchmark(self):
-        """Terminate the running benchmark process if active."""
-        if self._process and self._process.poll() is None:
-            self._logger.warning("Terminating benchmark process due to vLLM failure")
+        """Terminate the running benchmark process and its process group if active."""
+        # Try to use stored PID/PGID first, in case process handle is gone
+        pid = (
+            self._process_pid
+            if self._process_pid
+            else (self._process.pid if self._process else None)
+        )
```

Comment:
This was just hard to reason about, maybe a simpler change?

```suggestion
        pid = self._process_pid
        if pid is None:  # first check, if PID is still none after this we will return
            pid = self._process.pid if self._process else None
```